lib/module.js

```diff
@@ -605,11 +609,11 @@ Module.prototype.load = function(filename) {
 // `exports` property.
 Module.prototype.require = function(id) {
   if (typeof id !== 'string') {
-    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'id', 'string', id);
+    throw new ERR_INVALID_ARG_TYPE('id', 'string', id);
   }
   if (id === '') {
-    throw new errors.Error('ERR_INVALID_ARG_VALUE',
-                           'id', id, 'must be a non-empty string');
+    throw new ERR_INVALID_ARG_VALUE.RangeError('id', id,
+                                               'must be a non-empty string');
```

Comment:
To me it seems like this is a `TypeError`, not a `RangeError`. Don't you think so?

Reply:
The type is checked just before. I don't feel strongly about it but it seemed right to throw a RangeError: the must be greater than zero

Reply:
Well, I guess it is somewhat of a edge case. I do not feel strongly about it either.

Reply:
I think usually we just go with `ERR_OUT_OF_RANGE` for this sort of generic range errors..probably makes more sense if it's `ERR_INVALID_ARG_VALUE.Error` on `id`, or a `ERR_OUT_OF_RANGE` on `id.length`

Reply:
I changed it to a `TypeError`. @joyeecheung are you OK with that?